Improve user-friendliness

dobraczka opened the pull request:
Address #1

dobraczka committed:
Removed unused dependencies

dobraczka committed:
Use poetry

- Transform Pipfile to new pyproject.toml
- Update Readme with new instructions

dobraczka committed:
Forgot to add some changes to last commit

dobraczka committed:
Remove old type analysis code

dobraczka committed:
Remove old type analysis test

dobraczka committed:
Removed tests for non-existing code

dobraczka committed:
Started data downloading implementation

- src.dataset.dataset.DataSet contains new method download_and_unzip
- made corresponding adaptations for openea class
- wrote test class

dobraczka commented on src/dataset/openea_dataset.py:
```diff
@@ -29,7 +33,6 @@ def __init__(self, data_folder: str, division: str, args_path: str):
         )
         self._name = data_folder.split("/")[-2] + "/" + division[:-1]
 
-    @abstractmethod
```
This should not have been here right? Because it is actually implemented...

jonathanschuchart replied:
hmm, odd. No, I guess it shouldn't. Especially since OpenEaDataset is a concrete class

dobraczka committed:
Use local dataset path

dobraczka committed:
Generalized download function

dobraczka committed:
Added ipdb as dev dependency for debugging

dobraczka committed:
Use preformatted version of CSV datasets

- Uploaded formatted version to speicherwolke
- Adapted CsvDataset code to be effectively the same as OpenEaDataset

dobraczka committed:
Adadpted movie dataset for downloading and processing

dobraczka committed:
Dataset adaptations

- Use keyworded function call
- Adapted default train val ratio
- Use Dataset function to split Movie datasets dynamically because of
  innerlinks

dobraczka committed:
Use new classes in run config

dobraczka committed:
Adapted README

dobraczka commented on src/dataset/csv_dataset.py:
```diff
@@ -18,46 +13,34 @@ class CsvType(Enum):
-            mode=model.args.alignment_module,
-            ordered=model.args.ordered,
+    _download_urls = [
+        "https://speicherwolke.uni-leipzig.de/index.php/s/ro25Tb4WdB6Y4iJ/download"
```
This was easier  than to transform them and since its only ~20MB it is no problem

dobraczka commented on src/dataset/dataset.py:
```diff
@@ -102,9 +102,9 @@ def __init__(
         rnd: random.Random,
         labelled_pairs: List[Tuple[int, int, int]],
         labelled_val_pairs: List[Tuple[int, int, int]] = None,
-        val_ratio: Optional[float] = 0.2,
+        val_ratio: Optional[float] = 0.1,
         labelled_test_pairs: List[Tuple[int, int, int]] = None,
-        test_ratio: Optional[float] = 0.1,
+        test_ratio: Optional[float] = 0.7,
```
Why was the val and test ratio default different than in OpenEA? I changed them. I hope this is fine.

jonathanschuchart replied:
This must have happened on accident by a test...

dobraczka commented on src/dataset/scads_dataset.py:
```diff
@@ -17,17 +17,17 @@ def __init__(self, data_folder: str, division: str, args_path: str):
+            kg1=self._kgs.kg1,
+            kg2=self._kgs.kg2,
+            rnd=random.Random(),
+            labelled_pairs=train_links + valid_links + test_links,
```
I now let the dataset class calculate the split, because it was not correct with the removal of the inner links

jonathanschuchart replied:
Make sure the random class gets a fixed seed so the splits are reproducible

dobraczka commented:
Since the different dataset classes previously were used to get the data in the right format and I had the data already in the right format all the dataset classes get their data in OpenEA format and they mainly provide a way to gather the data

dobraczka commented on src/dataset/scads_dataset.py:
```diff
@@ -1,100 +1,80 @@
+                        "data", "MovieGraphBenchmark", "src", "create_graph.py"
+                    ),
+                ]
+            )
```
This was the simplest way to make this work without having to mess with the pythonpath etc.

dobraczka committed:
Use seed for random in scads dataset

dobraczka committed:
Merge branch 'main' into user-friendliness-#1
